### ISSUE TITLE
Why distance transform not used for Room detection?

### ISSUE THREAD

**Issue by shubhamwagh** [grebtsew/FloorplanToBlender3d#2]
Hi! May I know if you have tried distance transform for room detection? I was wondering if that can improve room detection result.

**Comment by grebtsew** [comment]
Hi,

Yes, when I added the room detection I tried a couple of ways of doing it. My original idea was to use  Image Segmentation with Distance Transform. I also looked at template matching, building my own neural network and shape detection. This was about six months ago and I don't really remember why I left the idea. Perhaps I wanted to move on with other projects and felt like the implementation already reached a "proof of concept" level, and therefore copied a working solution from stackoverflow. 

The room and detail (door & window) detections in this implementation needs some work. I will be out traveling now for about two weeks but when I get back I can take a look at it.

Cheers!
